asg/src/reducer/monoid/tuple.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+use super::*;
+
+// macro for implementing n-ary tuple functions and operations
+macro_rules! tuple_monoid_impls {
+    ($(
+        $Tuple:ident {
```

Comment:
Unnecessary input

Reply:
I didn't quite understand this. Could you elaborate?

Reply:
Sure, the meta-variable `$Tuple` isn't actually used in the macro expansion.